linux/utils/add_repo_from_baseurl.yml

```diff
@@ -1,31 +1,37 @@
-# Copyright 2021-2022 VMware, Inc.
+# Copyright 2021-2023 VMware, Inc.
 # SPDX-License-Identifier: BSD-2-Clause
 ---
 # Add a package repository with given baseurl
 # Parameter
 #   repo_name: (Required)The repository name
-#   repo_baseurl: (Required)The base url of the repository or the local repo dir
+#   repo_baseurl: (Required)The base url of the repository or the local repository dir
 #   repo_mirrorlist: (Optional)The URL to a mirror list of the repository, which is used for RHEL-like OS.
 #   repo_description: (Optional)The description of the repository
 #   gpg_check: (Optinoal)True to check GPG key
 #   gpg_key_path: (Optional)The GPG key path
 
-- name: "Check repo name and baseurl parameters are valid"
+- name: "Check repository name and baseurl parameters are valid"
   ansible.builtin.assert:
     that:
       - repo_name is defined and repo_name
       - (repo_baseurl is defined and repo_baseurl) or (repo_mirrorlist is defined and repo_mirrorlist)
     fail_msg: "Invalid parameters: repo_name={{ repo_name | default('undefined') }}, repo_baseurl={{ repo_baseurl | default('undefined') }}, repo_mirrorlist = {{ repo_mirrorlist | default('undefined') }}"
 
-- name: "Set default GPG key"
-  ansible.builtin.set_fact:
-    gpg_key_path: |-
-      {%- if guest_os_ansible_distribution == "RedHat" -%}file:///etc/pki/rpm-gpg/RPM-GPG-KEY-redhat-release
-      {%- elif guest_os_ansible_distribution == "CentOS" -%}file:///etc/pki/rpm-gpg/RPM-GPG-KEY-centosofficial
-      {%- elif guest_os_ansible_distribution == "Rocky" -%}file:///etc/pki/rpm-gpg/RPM-GPG-KEY-rockyofficial
-      {%- elif guest_os_ansible_distribution == "OracleLinux"  -%}file:///etc/pki/rpm-gpg/RPM-GPG-KEY-oracle
-      {%- elif guest_os_ansible_distribution == "VMware Photon OS"  -%}file:///etc/pki/rpm-gpg/VMWARE-RPM-GPG-KEY
-      {%- else -%}{%- endif -%}
+- name: "Set default RPM GPG key"
+  block:
+    - name: "Get default RPM GPG key file"
+      include_tasks: get_rpm_gpg_key_file.yml
+      when: guest_rpm_gpg_key_path is undefined or not guest_rpm_gpg_key_path
+
+    - name: "Set default RPM GPG key"
+      ansible.builtin.set_fact:
+        gpg_key_path: "file://{{ guest_rpm_gpg_key_path }}"
+      when: guest_rpm_gpg_key_path
+
+    - name: "Skip GPG check as there is no RPM GPG key found"
```

Comment:
Here need to fail since `gpg_check` is set but not found GPG key?

Reply:
Fixed.